tool/tsh/tsh.go

```diff
@@ -377,6 +377,9 @@ func (e *exitCodeError) Error() string {
 }
 
 func main() {
+	// Allow prompt to restore terminal state on exit.
+	defer prompt.NotifyExit()
```

Comment:
does this do anything when we `os.Exit` below?

Reply:
Good point, moved.